src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithDictionary.php

```diff
@@ -24,12 +24,23 @@ protected function getDictionaryKey($attribute)
             }
 
             if ($attribute instanceof UnitEnum) {
-                return $attribute instanceof BackedEnum ? $attribute->value : $attribute->name;
+                return $this->getEnumValue($attribute);
             }
 
             throw new InvalidArgumentException('Model attribute value is an object but does not have a __toString method.');
         }
 
         return $attribute;
     }
+
+    /**
+     * Get a value from an enum case if it is backed, otherwise use the case name.
+     *
+     * @param  UnitEnum  $attribute
+     * @return mixed
```

Comment:
```suggestion
     * @return string | int
```
Currently Enums can only be backed by a `string` or an `int` so we can be more precise in the return type here I think.